src/Parser.js

```diff
@@ -511,13 +547,13 @@ export default class Parser {
             // begin...end is similar to left...right
             const begin = this.parseGivenFunction(start);
             const envName = begin.value.name;
-            if (!environments.has(envName)) {
+            if (!environments.hasOwnProperty(envName)) {
```

Comment:
I got rid of the `has`/`get` methods from before b/c `flow` doesn't know how to analyze checks using those methods.

Reply:
That might of have fixed it, but this also brings `environments` in line with `functions`.